src/message.rs

```diff
@@ -28,15 +31,129 @@ fn _message_prettify(message: CString, comment_char: Option<u8>) -> Result<Strin
     Ok(ret.as_str().unwrap().to_string())
 }
 
+/// Collection of trailer key–value pairs.
+///
+/// Use `iter()` to get access to the values.
+pub struct MessageTrailers<'pair> {
+    raw: raw::git_message_trailer_array,
+    _marker: marker::PhantomData<&'pair c_char>,
+}
+
+impl<'pair> MessageTrailers<'pair> {
+    fn new() -> MessageTrailers<'pair> {
+        crate::init();
+        unsafe {
+            Binding::from_raw(&mut raw::git_message_trailer_array {
+                trailers: ptr::null_mut(),
+                count: 0,
+                _trailer_block: ptr::null_mut(),
+            } as *mut _)
+        }
+    }
+    /// Create a borrowed iterator.
+    pub fn iter(&'pair self) -> MessageTrailersIterator<'pair> {
+        MessageTrailersIterator {
+            trailers: self,
+            counter: 0,
+        }
+    }
+    /// The number of trailer key–value pairs.
+    pub fn len(&self) -> usize {
+        self.raw.count
+    }
+}
+
+impl<'pair> Drop for MessageTrailers<'pair> {
+    fn drop(&mut self) {
+        unsafe {
+            raw::git_message_trailer_array_free(&mut self.raw);
+        }
+    }
+}
+
+impl<'pair> Binding for MessageTrailers<'pair> {
+    type Raw = *mut raw::git_message_trailer_array;
+    unsafe fn from_raw(raw: *mut raw::git_message_trailer_array) -> MessageTrailers<'pair> {
+        MessageTrailers {
+            raw: *raw,
+            _marker: marker::PhantomData,
+        }
+    }
+    fn raw(&self) -> *mut raw::git_message_trailer_array {
+        &self.raw as *const _ as *mut _
+    }
+}
+
+struct Trailer<'pair> {
+    key: *const c_char,
+    value: *const c_char,
+    _marker: marker::PhantomData<&'pair c_char>,
+}
+
+impl<'pair> Trailer<'pair> {
+    fn to_str_tuple(self) -> (&'pair str, &'pair str) {
+        unsafe {
+            let key = CStr::from_ptr(self.key).to_str().unwrap();
+            let value = CStr::from_ptr(self.value).to_str().unwrap();
+            (key, value)
+        }
+    }
+}
+
+/// A borrowed iterator.
+pub struct MessageTrailersIterator<'pair> {
+    trailers: &'pair MessageTrailers<'pair>,
+    counter: usize,
```

Comment:
One thing I might recommend is to change this to `Range<usize>` so that way `next()` is easy to implement.

Reply:
Never heard of, but I like it!

See rev. e94008a00fd